pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -729,13 +742,23 @@ func (p *PriorityQueue) movePodsToActiveOrBackoffQueue(podInfoList []*framework.
 			gated := pInfo.Gated
 			if added, _ := p.addToActiveQ(pInfo); added {
 				klog.V(5).InfoS("Pod moved to an internal scheduling queue", "pod", klog.KObj(pInfo.Pod), "event", event, "queue", activeQName)
+				pInfo.MoveRequestCycle = p.schedulingCycle
 				activated = true
 				metrics.SchedulerQueueIncomingPods.WithLabelValues("active", event.Label).Inc()
 				p.unschedulablePods.delete(pod, gated)
 			}
 		}
 	}
-	p.moveRequestCycle = p.schedulingCycle
+
+	// Besides setting MoveRequestCycle above for all pods which got moved,
+	// we also need to set it for pods that are being processed because
+	// AddUnschedulableIfNotPresent might get called for them, and in
+	// AddUnschedulableIfNotPresent we need to know whether events where
+	// observed while processing them.
+	for pInfo := range p.activePods {
+		pInfo.MoveRequestCycle = p.schedulingCycle
+	}
```

Comment:
My _gut_-feeling is that this pointer-based `activePods` map can be  optimized, but it's good enough for the current version. Let's maybe add a todo here?

Reply:
TODO added.

What would you use instead?

Reply:
I _was_ thinking about something like a bookmark thing, or a (doubly) linkedlist. I'm feeling that we only need to do a partial update to achieve what we want. But maybe an overkill 🤷